core/remotes/docker/config/hosts.go

```diff
@@ -293,6 +294,39 @@ func hostDirectory(host string) string {
 	return host
 }
 
+// hostMatchWildcardPrefix scans the root directory for any directories beginning w/ a '_'
+// The rest of the directory name is used to check the current host for a match. If a match is found it is appended to wildcardHosts.
+// The returning list is ordered by match length in descending order, meaning if "_.registry.io" and "_.io" are present, than "_.registry.io" will take precedence over "_.io"
```

Comment:
Why not implicitly search this way without a special `_.`?

Reply:
The `_.` was recommended in the previous PR here, https://github.com/containerd/containerd/pull/7758#discussion_r1049115631, by @dmcgowan. 

In this case the `_.` is to differentiate between the usage of `_default`. If I were to use it implicitly than I wouldn't be able to differentiate between `_default` and `_.default`.